microsoft/testsuites/power/power.py

```diff
@@ -0,0 +1,680 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+import time
+from datetime import date
+
+from assertpy import assert_that
+
+from lisa import (
+    Environment,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    schema,
+    search_space,
+)
+from lisa.base_tools import Cat, Uname
+from lisa.features import StartStop
+from lisa.operating_system import Redhat, Suse
+from lisa.sut_orchestrator.azure.tools import VmGeneration
+from lisa.testsuite import simple_requirement
+from lisa.tools import Blkid, Dmesg, Echo, Fdisk, Sed, Swap, SwapOn
+from lisa.util import (
+    LisaException,
+    SkippedException,
+    find_patterns_in_lines,
+    get_matched_str,
+)
+from lisa.util.logger import Logger
+from lisa.util.shell import wait_tcp_port_ready
+
+# matching 4-digit year between whitespace
+# or at the beginning of a line
+# example: 2022 from '2022-02-04 11:38:06.949136+0000'
+# or 2018 from 'Fri 07 Sep 2018 11:26:52 AM CEST .838868 seconds'
+# will be matched
+_YEAR_PATTERN = re.compile("(\s\d{4}\s)|(^\d{4})")
+
+
+def _check_vm_feature_support_status(min_support_version: str, node: Node) -> bool:
+    if not min_support_version:
+        raise LisaException("No min support kernel version provided!")
+
+    kernel_version = node.tools[Uname].get_linux_information().kernel_version
+    if kernel_version > min_support_version:
+        return True
+    elif kernel_version < min_support_version:
+        return False
+
+    return True
+
+
+def _get_year(node: Node) -> str:
+    hard_ware_clock = node.execute(
+        "hwclock",
+        shell=True,
+        sudo=True
+    ).stdout
+    # Ubuntu 16.4.0 format: Fri 07 Sep 2018 11:26:52 AM CEST .838868 seconds
+    # other hwclock format: '2022-02-04 11:38:06.949136+0000'
+    year = get_matched_str(hard_ware_clock, _YEAR_PATTERN)
+    return year
+
+
+def _found_sys_log(node: Node, message: str) -> bool:
+    if node.shell.exists('/var/log/messages'):
+        _ret = node.tools[Cat].read_with_filter(
+            "/var/log/messages",
+            message,
+            sudo=True
+        )
+    elif node.shell.exists('/var/log/syslog'):
+        _ret = node.tools[Cat].read_with_filter(
+            "/var/log/syslog",
+            message,
+            sudo=True
+        )
+    else:
+        _ret = node.execute(
+            f'dmesg | grep -i {message}',
+            shell=True,
+            sudo=True
+        ).stdout
+    if message in _ret:
+        return True
+    else:
+        return False
+
+
+def _is_vm_running(node: Node, log: Logger, timeout: int = 300) -> bool:
+    is_ready, tcp_error_code = wait_tcp_port_ready(
+        node.public_address,
+        node.public_port,
+        log=log,
+        timeout=timeout,
+    )
+    return is_ready
+
+
+def _recover_files(node: Node, log: Logger, files_and_contents: dict) -> None:
+    for file in files_and_contents:
+        node.tools[Echo].write_to_file(
```

Comment:
Some file may be big. It's better to make a backup at the same path and recover it. It's faster.

Reply:
what's the prefer method we use to backup? is it 
`mv <file path> <file path>.bak`
and for restore
`rm <file path>`
`mv <file path>.bak <file path`

besides, the sed functions will create .bak files which will replace the existing one. Should we put backup files in somewhere else. e.g. /tmp ?

Reply:
Your approach is right. It's better to use a unique backup file name for the current tests, so it won't be overwritten by other stuffs suddenly.